tests/common.go

```diff
@@ -4804,27 +4805,42 @@ func ValidatePoolRebalance() error {
 		return err
 	}
 
-	var pools map[string]*opsapi.StoragePool
-	pools, err = Inst().V.ListStoragePools(metav1.LabelSelector{})
-	log.FailOnError(err, "error getting pools list")
+	nodePoolsToValidate := make([]node.StoragePool, 0)
+	if poolID != -1 {
+		for _, p := range stNode.StoragePools {
+			if p.ID == poolID {
+				nodePoolsToValidate = append(nodePoolsToValidate, p)
+				break
+			}
+		}
+	} else {
+		//A new pool might be created due to add drive,hence 2 min wait for pool to associate with node
+		time.Sleep(2 * time.Minute)
```

Comment:
Can we not use `task.DoRetryWithTimeout` here?

Reply:
retry is not required here. we are waiting so that when refresh the node registry, new pools is reflected.
PX will take some time to tag new drive added to tag it as a pool